tests/partition/test_leiden.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Tuple
+
+import unittest
+import networkx as nx
+
+from graspologic.partition import HierarchicalCluster, hierarchical_leiden, leiden
+
+from tests.utils import data_file
+
+
+class TestHierarchicalCluster(unittest.TestCase):
+    def test_from_native(self):
+        with self.assertRaises(TypeError):
+            HierarchicalCluster.from_native("foo")
+
+        # note: it is impossible to create a native instance of a HierarchicalCluster.  We will
+        # test from_native indirectly through calling graspologic.partition.hierarchical_leiden()
+
+    def test_final_hierarchical_clustering(self):
+        with self.assertRaises(TypeError):
+            HierarchicalCluster.final_hierarchical_clustering("I am not a list")
+
+        hierarchical_clusters = [
+            HierarchicalCluster("1", 0, None, 0, False),
+            HierarchicalCluster("2", 0, None, 0, False),
+            HierarchicalCluster("3", 0, None, 0, False),
+            HierarchicalCluster("4", 1, None, 0, True),
+            HierarchicalCluster("5", 1, None, 0, True),
+            HierarchicalCluster("1", 2, 0, 1, True),
+            HierarchicalCluster("2", 2, 0, 1, True),
+            HierarchicalCluster("3", 3, 0, 1, True),
+        ]
+
+        expected = {
+            "1": 2,
+            "2": 2,
+            "3": 3,
+            "4": 1,
+            "5": 1,
+        }
+        self.assertEqual(
+            expected,
+            HierarchicalCluster.final_hierarchical_clustering(hierarchical_clusters),
+        )
+
+
+def _create_edge_list() -> List[Tuple[str, str, float]]:
+    edges = []
+    with open(data_file("large-graph.csv"), "r") as edges_io:
```

Comment:
is a test this big necessary? Not sure how long it takes/how big that file is

Reply:
About 2.56s.  Compared to tests/test_latentdistributiontest.py @ 160.19s and tests/test_ptr.py @ 2.22s.

Pretty negligible, to be honest.  I can go smaller but I actually don't have a graph smaller than this - if you have any laying around we can commit in some form that's fine, I just have to update these and the modularity tests